### ISSUE TITLE
[Feature Request] Add ability to assign a scene number to any line

### ISSUE THREAD

**Issue by hyuri** [story-apps/starc#172]
Sometimes we have a scene briefly interwoven into another. It is written all contained into one scene, under one scene header, but, in terms of production, it's really two scenes that end up treated as two independent scenes during production, and filming. An example is a brief flashback contained within a scene. Even though it is part of one scene, it will be treated and filmed as its own separate scene during production.

So it would be useful to be able to assign a scene number to any lines. Or at least line types such as "action", "shot" and "unformatted".

**Reference:**
https://johnaugust.com/2015/short-cut-aways-and-the-value-of-back-to

From the reference:
```
In production
When it comes time to make the movie, everything needs a scene number. We generally think of scene numbers going with scene headers, but the reality is that anything can have a number attached, including the italicized action lines above.
```

**Comment by dimkanovikov** [comment]
Why not to add one more scene heading here? It’s okay for shooting scripts as I know.

**Comment by hyuri** [comment]
Yes, usually for the shooting script you rearrange a few things so each moment that needs to be shot as a separate scene will have its own scene [header].

I was just thinking we could simplify the process. Let's call these lines marked with scene numbers "Subscenes" — a scene contained within another scene.

Maybe, instead of assigning a scene number these lines, we could simply mark them as "Subscene", and Starc does the subscene numbering automatically. So one subscene contained within scene "1" would become scene "A1"; a second subscene in scene "1" would become scene "B1" and so on.

Going even further, perhaps Starc could even have a display option "Separate Subscenes" or even have a Shooting Module that would take these subcenes and display them as separate scenes after the scenes that contain them.

**So, for example (with subscene FLASHBACK marked with an asterisk):**

```
1. INT. JACK'S HOUSE, LIVING ROOM - DAY
   Jack freezes. Stares into the void.

*1 FLASHBACK

*1 Jack floats in the middle of a river. Josh emerges out of the water and pulls him down.

   BACK TO SCENE

   Jack unfreezes. Blinks. Shakes his head.

                     JACK
                No!
```

**Would be displayed as:**

```
1. INT. JACK'S HOUSE, LIVING ROOM - DAY
   Jack freezes. Stares into the void.

   Jack unfreezes. Blinks. Shakes his head.

                     JACK
                No!

A1. FLASHBACK

    Jack floats in the middle of a river. Josh emerges out of the water and pulls him down.

```

What do you think?

**Comment by dimkanovikov** [comment]
We talk about it in terms of preproduction, or writing? What value this feature brings to users?

**Comment by hyuri** [comment]
Pre-production.

From the link https://johnaugust.com/2015/short-cut-aways-and-the-value-of-back-to:
> **In production**
> When it comes time to make the movie, everything needs a scene number. We generally think of scene numbers going with scene headers, but the reality is that anything can have a number attached, including the italicized action lines above.
> 
> There are different philosophies for how to number flashback scenes, but my preference would be to keep the copy room scene as a single scene number (e.g. 34) and group together all of the examination room scenes as a sequence (e.g. A900, B900, C900). This way, the copy room scene doesn’t get divided across a few strips, potentially confusing everyone.
> 
> Numbering scenes is a conversation to have with the director, A.D. and line producer. It’s a luxury problem, because it means your movie is getting made.

**Comment by dimkanovikov** [comment]
Hm... if it will be a separate scene, then it should have not only its own number, but also its own location, time, cast list, etc., so making an action line as a scene looks like additional headache instead of improvement :) 

So, I think it should be reformatted something like this 

```
1-1. INT. JACK'S HOUSE, LIVING ROOM - DAY
     Jack freezes. Stares into the void.

 F1. INT. RIVER - DAY

     Jack floats in the middle of a river. Josh emerges out of the water and pulls him down.

1-2. INT. JACK'S HOUSE, LIVING ROOM - DAY

     Jack unfreezes. Blinks. Shakes his head.

                     JACK
                No!
```

I really don't understand, what benefit can provide ability to set numbers to some of the script lines.

**Comment by hyuri** [comment]
Right now, the 1st A.D. would manually take this scene:

```
INT. JACK'S HOUSE, LIVING ROOM - DAY
Jack freezes. Stares into the void.

FLASHBACK

Jack floats in the middle of a river. Josh emerges out of the water and pulls him down.

BACK TO SCENE

Jack unfreezes. Blinks. Shakes his head.

                     JACK
                No!
```

**Would number the scene something like:**

```
1. INT. JACK'S HOUSE, LIVING ROOM - DAY
   Jack freezes. Stares into the void.

A1. FLASHBACK

A1. Jack floats in the middle of a river. Josh emerges out of the water and pulls him down.

    BACK TO SCENE

    Jack unfreezes. Blinks. Shakes his head.

                      JACK
                 No!
```

**And then manually create a new version of the script, with two scenes, looking like this:**

```
1. INT. JACK'S HOUSE, LIVING ROOM - DAY
   Jack freezes. Stares into the void.

   Jack unfreezes. Blinks. Shakes his head.

                     JACK
                No!

A1. EXT. RIVER - DAY [FLASHBACK]

    Jack floats in the middle of a river. Josh emerges out of the water and pulls him down.

```

My suggestion of assigning scene numbers to any line would help the 1st A.D. in the numbering process, before creating the new version of the script with the scenes separated with their own scene headers. And my other suggestions (of actually using the numbering of lines to automatically create this new version of the script — the shooting script) would help cutting a few steps, automating the process and making it much faster.

In the end, the 1st A.D. would end up with the same result as if he had done it manually, but it would be much easier and faster.
Just like the Script Breakdown is something the 1st A.D. will do manually but Starc offers the Breakdown module to speed things up.

In fact, I think it would be better to have this feature inside the Breakdown module. Makes more sense. Since it's part of the 1st A.D.'s process.
